maro/data_lib/dump_csv_converter.py

```diff
@@ -0,0 +1,201 @@
+
+        folder_path = Path(self._foldername)
+        if folder_path.exists():
+            return
+        os.mkdir(self._foldername)
+
+    @property
+    def dump_folder(self):
+        return self._foldername
+
+    @property
+    def current_serial(self):
+        return self._serial
+
+    @property
+    def scenario_name(self):
+        return self._scenario_name
+
+    def reset_folder_path(self):
+        self._generate_new_folder(self._parent_path)
+        self._serial = 0
+
+    def get_new_snapshot_folder(self):
+        folder = os.path.join(self._foldername, self._prefix + str(self._serial))
+        os.mkdir(folder)
+        self._serial = self._serial + 1
+        self._last_snapshot_folder = folder
+        return folder
+
+    def process_data(self, filesource: str):
+        for cur_dir, dirs, files in os.walk(self._last_snapshot_folder):
+            for file in files:
+                if file.endswith(".meta"):
+                    col_info_dict = self.get_column_info(os.path.join(cur_dir, file))
+                    data = np.load(os.path.join(cur_dir, file.replace(".meta", ".npy")))
+
+                    frame_idx = 0
+                    csv_data = []
+                    for frame in data:
+                        node_idx = 0
+                        file_name = file.replace(".meta", "")
+                        for node in frame:
+                            node_dict = {"frame_index": frame_idx, "name": file_name + "_" + str(node_idx)}
+                            col_idx = 0
+                            for key in col_info_dict.keys():
+                                if col_info_dict[key] == 1:
+                                    node_dict[key] = node[col_idx]
+                                else:
+                                    node_dict[key] = str(node[col_idx])
+                                col_idx = col_idx + 1
+
+                            node_idx = node_idx + 1
+                            csv_data.append(node_dict)
+
+                        frame_idx = frame_idx + 1
+
+                    dataframe = pd.DataFrame(csv_data)
+                    dataframe.to_csv(os.path.join(cur_dir, file.replace(".meta", ".csv")), index=False)
+
+        self.save_manifest_file(filesource)
+
+    def start_processing(self, filesource: str):
+        thread = threading.Thread(target=self.process_data, args=(filesource,))
+        thread.start()
+
+    def get_column_info(self, filename):
+        with open(filename, "r") as f:
+            columns = f.readline().strip()
+            elements = f.readline().strip()
+            f.close()
+
+        col_dict = {}
+        cols = str.split(columns, ",")
+        element_list = str.split(elements, ",")
+        i = 0
+        for col in cols:
+            col_dict[col] = element_list[i]
+            i = i + 1
+        return col_dict
+
+    def clear_raw_data(self):
+        for curDir, dirs, files in os.walk(self._foldername):
```

Comment:
same here

Reply:
same here with - if we do not need curDir and dirs in here, what about replacing it with '_'